src/aws/osml/model_runner/database/feature_table.py

```diff
@@ -121,28 +122,27 @@ def add_features(self, features: List[Feature], metrics: MetricsLogger = None):
                             total_encoded_length = 0
                             encoded_features = []
 
-                            # Check that we got a success response
-                            status_code = result.get("ResponseMetadata", {}).get("HTTPStatusCode")
-                            if status_code != 200:
-                                if isinstance(metrics, MetricsLogger):
-                                    metrics.put_metric(MetricLabels.FEATURE_UPDATE, 1, str(Unit.COUNT.value))
-                                    metrics.put_metric(MetricLabels.FEATURE_ERROR, 1, str(Unit.COUNT.value))
-                                logger.error("Unable to update feature table - HTTP Status Code: {}".format(status_code))
-                except Exception as err:
+                # Write the items to the table and check that it succeeded
+                if not self.batch_write_items(items):
                     if isinstance(metrics, MetricsLogger):
-                        metrics.put_metric(MetricLabels.FEATURE_UPDATE_EXCEPTION, 1, str(Unit.COUNT.value))
+                        metrics.put_metric(MetricLabels.FEATURE_UPDATE, 1, str(Unit.COUNT.value))
                         metrics.put_metric(MetricLabels.FEATURE_ERROR, 1, str(Unit.COUNT.value))
-                    logger.error("There was a problem adding features: {}".format(err))
-                    raise AddFeaturesException("Failed to add features for tile!") from err
+                    logger.error("Unable to push features to feature table!")
```

Comment:
Wrong logger.error, wouldn't this be: `logger.info("Successfully pushed features to the feature table!")`

Reply:
No - if the batch write operation returns false ( which leads to this block ) then that means some items were not written successfully.